modules/1_prepare/prepare.tf

```diff
@@ -145,6 +145,9 @@ then
     sudo subscription-manager register --username=${var.rhel_subscription_username} --password=${var.rhel_subscription_password} --force
     sudo subscription-manager refresh
     sudo subscription-manager attach --auto
+elif [ '${var.rhel_subscription_org}' != '' ]
+then
+   sudo subscription-manager register --activationkey=${var.rhel_subscription_activationkey} --org=${var.rhel_subscription_org} --force
```

Comment:
Should the following two commands be kept out of the if-then-else block ?
```
sudo subscription-manager refresh
sudo subscription-manager attach --auto
```

Reply:
How about unregistering the system during destroy?

Reply:
I don't have a definitive answer on that. My team has been using this method for years without issue. I also don't see a detach on destroy in this project for the username/password method.

Reply:
For now this is fine. And @zmc as you are already using this approach so will go with your word on not requiring `attach`.